services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidateSimpleFlowSpec.groovy

```diff
@@ -0,0 +1,517 @@
+            checkNoMeterValidation(it.dpId)
+        }
+    }
+
+    def "Able to get empty meter validate information from the intermediate switch(flow contain > 2 switches)"() {
+        when: "Create an intermediate-switch flow"
+        def flow = intermediateSwitchFlow()
+        flowHelper.addFlow(flow)
+        def flowPath = PathHelper.convert(northbound.getFlowPath(flow.id))
+
+        then: "The intermediate switch does not contain any information about meter"
+        checkNoMeterValidation(flowPath[1].switchId)
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that meter validate info is also deleted"
+        flowPath.each {
+            if (switchHelper.getDescription(it.switchId).contains("OF_13")) {
+                checkNoRuleMeterValidation(it.switchId)
+                checkNoMeterValidation(it.switchId)
+            }
+        }
+    }
+
+    def intermediateSwitchFlow(boolean ensureAltPathsExist = false, boolean getAllPaths = false) {
+        def flowWithPaths = getFlowWithPaths(3, Integer.MAX_VALUE, ensureAltPathsExist ? 1 : 0)
+        return getAllPaths ? flowWithPaths : flowWithPaths[0]
+    }
+
+    def getFlowWithPaths(int minSwitchesInPath, int maxSwitchesInPath, int minAltPaths) {
+        def allFlowPaths = []
+        def switches = topology.getActiveSwitches()
+        def switchPair = [switches, switches].combinations().findAll {
+            src, dst -> src != dst
+        }.unique {
+            it.sort()
+        }.find { src, dst ->
+            allFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allFlowPaths.size() > minAltPaths && allFlowPaths.min {
+                it.size()
+            }.size() in (minSwitchesInPath..maxSwitchesInPath)
+        } ?: assumeTrue("No suiting switches found", false)
+
+        return [flowHelper.randomFlow(switchPair[0], switchPair[1]), allFlowPaths]
+    }
+
+    @Memoized
+    List<Switch> getNonCentecSwitches() {
+        topology.activeSwitches.findAll { !it.centec && it.ofVersion == "OF_13" }
+    }
+
+    @Memoized
+    List<Switch> getCentecSwitches() {
+        topology.getActiveSwitches().findAll { it.centec }
+    }
+
+    @Memoized
+    List<Switch> getSwitchNotSupportedMeter() {
+        topology.activeSwitches.findAll { it.ofVersion == "OF_12" }
+    }
+
+    List<Integer> getCreatedMetersId(switchId, defaultMeters) {
+        northbound.getAllMeters(switchId).meterEntries.findAll {
+            !defaultMeters.meterEntries*.meterId.contains(it.meterId)
+        }.collect { it.meterId }
+    }
+
+    List<Integer> getCreatedCookies(switchId, createdMetersId) {
+        def createdCookies = []
+        createdMetersId.each { id ->
+            northbound.getSwitchRules(switchId).flowEntries.each {
+                if (it.instructions.goToMeter == id) {
+                    createdCookies << it.cookie
+                }
+            }
+        }
+        return createdCookies
```

Comment:
Your implementation calls getSwitchRules for every 'createdMetersId' multiple times for no reason. Should be called only once.
```
return northbound.getSwitchRules(switchId).flowEntries.collect { rule ->
   if(rule.instructions.goToMeter in createdMetersId) {
       rule.cookie
   }
}
```
OR
```
return northbound.getSwitchRules(switchId).flowEntries.findAll { 
     it.instructions.goToMeter in createdMetersId
}*.cookie
```

Reply:
By the way, I don't think that this is the right way to obtain the 'created cookies'.
created rules = all rules - default rules
In this method you operate with created meter ids. Is this intentional? I think its wrong, unless you explain the idea for me

Reply:
NOT FIXED

Reply:
Fixed